py/torch_tensorrt/dynamo/conversion/impl/normalization/ops.py

```diff
@@ -189,26 +191,97 @@ def native_group_norm(
 
     weight = get_trt_tensor(ctx, weight, f"{name}_weight")
     bias = get_trt_tensor(ctx, bias, f"{name}_bias")
-    if tuple(reshaped_input.shape) != tuple(weight.shape):
-        weight = impl.slice.expand(
-            ctx,
-            target,
-            source_ir,
-            f"{name}_expand_weight",
-            weight,
-            reshaped_input.shape,
-        )
-    if tuple(reshaped_input.shape) != tuple(bias.shape):
-        bias = impl.slice.expand(
-            ctx, target, source_ir, f"{name}_expand_bias", bias, reshaped_input.shape
-        )
+    weight_bias_shape = (1, C) + (1,) * (len(input.shape) - 2)
+
     dims = list(range(1, len(input.shape)))
-    axes = get_axes_for_reduce_op(dims)
-    group_norm = ctx.net.add_normalization(reshaped_input, weight, bias, axes)
-    group_norm.epsilon = eps
-    group_norm.compute_precision = input.dtype
-    set_layer_name(group_norm, target, f"{name}_group_norm", source_ir)
-    output = group_norm.get_output(0)
+
+    # E[X]
+    mean_trt = impl.reduce.mean(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_mean",
+        reshaped_input,
+        dims,
+        True,
+    )
+
+    mean_trt = impl.slice.expand(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_expand_mean_trt",
+        mean_trt,
+        reshaped_input.shape,
+    )
+
+    # X - E[X]
+    sub_trt = impl.elementwise.sub(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_sub",
+        reshaped_input,
+        mean_trt,
+    )
+
+    # variance
+    pow_trt = get_trt_tensor(ctx, 2, f"{name}_power", np.float32)
+    pow_var = impl.elementwise.pow(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_pow",
+        sub_trt,
+        pow_trt,
+    )
+
+    var_trt = impl.reduce.mean(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_mean_var",
+        pow_var,
+        dims,
+        True,
+    )
+
+    var_trt = impl.slice.expand(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_expand_var_trt",
+        var_trt,
+        reshaped_input.shape,
+    )
+
+    eps_trt = get_trt_tensor(ctx, eps, f"{name}_eps", np.float32)
+    add_trt = impl.elementwise.add(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_add",
+        var_trt,
+        eps_trt,
+    )
+
+    sqrt_trt = impl.unary.sqrt(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_sqrt",
+        add_trt,
+    )
+
+    # y = (X - E[X]) / sqrt((var + eps))
+    output = impl.elementwise.div(
+        ctx,
+        target,
+        source_ir,
+        f"{name}_div",
+        sub_trt,
+        sqrt_trt,
+    )
 
```

Comment:
Wanted to clarify if this div would require any mode, eg: trunc? Are the data types always compatible with the output types?

Reply:
I don't think so. The previous implementation from Evan did not include any mode.

Reply:
just FYI: we have `impl.elementwise.floor_divide` and `impl.elementwise.trunc_div` helper functions, if needed.